src/main/java/net/mcreator/ui/dialogs/imageeditor/ResizeDialog.java

```diff
@@ -43,29 +45,40 @@ public ResizeDialog(MCreator window, Canvas canvas, Layer layer, VersionManager
 		JSpinner width = new JSpinner(new SpinnerNumberModel(layer.getWidth(), 0, 10000, 1));
 		JSpinner height = new JSpinner(new SpinnerNumberModel(layer.getHeight(), 0, 10000, 1));
 		JCheckBox type = new JCheckBox();
+		JCheckBox affectCanvas = new JCheckBox();
 
-		JButton cancel = new JButton(UIManager.getString("OptionPane.cancelButtonText"));
 		JButton ok = L10N.button("action.common.resize");
+		JButton cancel = new JButton(UIManager.getString("OptionPane.cancelButtonText"));
 		ok.setBackground((Color) UIManager.get("MCreatorLAF.MAIN_TINT"));
 		ok.setForeground((Color) UIManager.get("MCreatorLAF.BLACK_ACCENT"));
 		getRootPane().setDefaultButton(ok);
 
 		GridBagConstraints layoutConstraints = new GridBagConstraints();
 
-		cancel.addActionListener(e -> setVisible(false));
-
 		ok.addActionListener(e -> {
 			layer.resize((int) width.getValue(), (int) height.getValue(), type.isSelected());
-			versionManager.addRevision(new Modification(canvas, layer));
+			if (affectCanvas.isSelected() && (canvas.getWidth() < layer.getWidth()

```

Comment:
What if the layer is not located in the top left corner? It won't always resize the layer even if needed. (eg. canvas is 10x10, layer is 8x8 and located offset by 2 on each axis. Resizing the layer to 9x9 should trigger canvas resize but it doesn't)

Reply:
@Defeatomizer this still needs to be addressed though :)

Reply:
This is a real thing, I'd need a couple of days (or more) to find the solution. Don't bother removing the _stale_ label: in case this PR gets closed, I'll reopen it when I fix this issue.
>**How would you implement it?**

I think there should be a way to move the canvas itself leaving all the layers at their place of the screen (after canvas is resized, of course).

Reply:
Take your time :)